docs/src/running_tests_locally.md

```diff
@@ -0,0 +1,23 @@
+# Running Tests Locally
+
+Mooncake.jl's test suite is fairly extensive. While you can use `Pkg.test` to run the test suite in the standard manner, this is not usually optimal in Mooncake.jl. When editing some code, you typically only want to run the tests associated with it, not the entire test suite.
+
+Mooncake's tests are organised as follows:
+1. Things that are required for most / all test suites are loaded up in `test/front_matter.jl`.
+1. The tests for something in `src` are located in an identically-named file in `test`. e.g. the unit tests for `src/rrules/new.jl` are located in `test/rrules/new.jl`.
+
+Thus, a workflow that I (Will) find works very well is the following:
+1. Ensure that you have Revise.jl and TestEnv.jl installed in your default environment.
+1. start the REPL, `dev` Mooncake.jl, and navigate to the top level of the Mooncake.jl directory.
+1. `using TestEnv, Revise`. Better still, load both of these in your `.julia/config/startup.jl` file so that you don't ever forget to load them.
+1. Run the following: `using Pkg; Pkg.activate("."); TestEnv.activate(); include("test/front_matter.jl");` to set up your environment.
```

Comment:
not a suggestion, just a anecdotal point from me: I configure VSCode to automatically do the `Pkg.activate()` for the package folder I open. But what you wrote is better as it assumes less.

Reply:
Thanks for letting me know -- this is good to know.